include/tvm/ir/source_map.h

```diff
@@ -127,6 +127,31 @@ class Span : public ObjectRef {
   TVM_DEFINE_OBJECT_REF_METHODS(Span, ObjectRef, SpanNode);
 };
 
+class MultiSpanNode : public SpanNode {
+ public:
+  Map<Span, ObjectRef> spans;
```

Comment:
I am not sure if it should be a dictionary, the span should be self-contained ideally and can be detached, so at best it should be an Array of Span to show the original tracking history

Reply:
the thing is, in addition to the Span I need the name of the Relax op it was attached to. Happy to report less stuff here, but I do need more than just what's in the span. Hence the map.

Reply:
Putting the object itself is more of an anti-pattern. You can however put in more string information into the fused span. Ideally span should not refer back to the IR fragement, this can cause things like cylic deps and other bad issues

Reply:
that's reasonable. i'll put the op then, unless someone has a more general suggestion.

Reply:
I think it should be something that is only related to span plus string in this particular case, since the main purpose of span is location tracking and nothing else

Reply:
changed to Map<String,Span>